web/app/js/components/DeploymentDetail.jsx

```diff
@@ -60,17 +64,24 @@ export default class DeploymentDetail extends React.Component {
 
     let urls = this.api.urlsForResource;
 
-    let podListFetch = this.api.fetchPods();
     let deployMetricsUrl = urls["deployment"].url(this.state.deploy).rollup;
     let upstreamRollupUrl = urls["upstream_deployment"].url(this.state.deploy).rollup;
     let downstreamRollupUrl = urls["downstream_deployment"].url(this.state.deploy).rollup;
 
-    let deployFetch = this.api.fetchMetrics(deployMetricsUrl);
-    let upstreamFetch = this.api.fetchMetrics(upstreamRollupUrl);
-    let downstreamFetch = this.api.fetchMetrics(downstreamRollupUrl);
+    this.requests = {
+      deploy: this.api.fetchMetrics(deployMetricsUrl),
+      upstreams: this.api.fetchMetrics(upstreamRollupUrl),
+      downstreams: this.api.fetchMetrics(downstreamRollupUrl),
+      podList: this.api.fetchPods()
+    };
```

Comment:
Minor nit, but I am good with shipping this as is.

Do you think it's worth while to have a function like`this.api.createRequestsHandler`  that creates an object similar to `this.requests`. The handler function could take in an array of urls and generate a `this.requests` object with a method called `cancel()` The `cancel()` method would then get all pending requests and cancel all outstanding promises. 

This would encapsulate
```javascript
 if (!_.isEmpty(this.requests)) {
+      _.each(this.requests, promise => promise.cancel());
+    }
```
so it's not repeated across the pages. Let me know if that makes any sense.

Reply:
I like this idea! I'm going to try moving the promises themselves to the ApiHelpers, I might move more things over later on.